### PR TITLE
[Hubert] Fix padding mask

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This PR is a copy of https://github.com/pytorch/fairseq/pull/3228 for Hubert. Wav2Vec2 had an issue with the padding mask which was solved in https://github.com/pytorch/fairseq/pull/3228. Hubert seems to have the same problem as Wav2vec2 (see: https://github.com/pytorch/fairseq/issues/3227) so that the padding mask should be corrected accordingly.

## What does this PR do?
Fixes https://github.com/pytorch/fairseq/issues/3227 for Hubert

## PR review    
@hikushalhere
@alexeib 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Hey @alexeib,

Is this PR merged actually? :-)

**Comment by alexeib** [comment]
hmm it might have gotten lost. lets see if i can rebase this and merge

File: fairseq/models/hubert/hubert.py [hunk 1]
```diff
@@ -397,16 +398,36 @@ def forward_targets(
...
         self, features: torch.Tensor, padding_mask: torch.Tensor,
     ) -> torch.Tensor:
-        extra = padding_mask.size(1) % features.size(1)
-        if extra > 0:
-            padding_mask = padding_mask[:, :-extra]
-        padding_mask = padding_mask.view(
-            padding_mask.size(0), features.size(1), -1
+        input_lengths = (1 - padding_mask.long()).sum(-1)
+        # apply conv formula to get real output_lengths
+        output_lengths = self._get_feat_extract_output_lengths(input_lengths)
```
Comment (wnhsu) [inline fairseq/models/hubert/hubert.py:421]: This needs to be changed to
```
torch.clamp(
  self._get_feat_extract_output_lengths(input_lengths),
  max=features.shape[1]
)
```
because the features may have been trimmed in `forward_targets()`

Comment (patrickvonplaten) [reply fairseq/models/hubert/hubert.py:421]: This probably needs to be changed in Wav2Vec2 then as well
